main_supcon.py

```diff
@@ -402,10 +437,35 @@ def train_category(train_loader, model, criterion, optimizer, epoch, opt):
 
     return losses.avg
 
+def f1_graph(epoch,question_number,model_name,weights_path,opt,question_dir,file_name,title):
```

Comment:
add type hints here. This function definition looks messy and has no documentation

Reply:
right now this function not needed its for validation and test fscore during test